sootup.callgraph/src/main/java/sootup/callgraph/GraphBasedCallGraph.java

```diff
@@ -75,15 +79,15 @@ protected GraphBasedCallGraph(
 
   @Override
   public void addMethod(@NonNull MethodSignature calledMethod) {
-    Vertex v = new Vertex(calledMethod);
+    Vertex v = new Vertex(calledMethod, vertexIdCounter++);
     addMethod(calledMethod, v);
   }
 
   protected void addMethod(@NonNull MethodSignature calledMethod, Vertex vertex) {
     if (containsMethod(calledMethod)) {
       return;
     }
-    graph.addVertex(vertex);
+    graph.addLabeledVertex(vertex);
```

Comment:
correct me if I'm wrong - but can't we pass in the MethodSignature itself? so we don't even need the Vertex class, anymore.